.github/workflows/knowledge_graph.py

```diff
@@ -185,21 +188,46 @@ def _check_md_file_contents(repo_root: Path, md_file: Path, ast: List[MdValue])
     return True
 
 
+@lru_cache()
+def _check_outbound_link(url: str):
+    # Following urls work correctly on a web browser but 404 when using python requests
+    whitelist = frozenset(["https://www.jessicahische.is/talkingtype"])
+    if url in whitelist:
+        return True
+
+    try:
+        url_status_code = requests.get(url).status_code
```

Comment:
I had suggested `requests.head` indeed, as that should make it faster and less bandwidth, possibly therefore cheaper. Regarding redirects, I noticed one of the URLs I fixed was redirecting, so I think a WARN result on the detection of redirects, and a FAIL if the redirect 4xx's, would be ideal.

Reply:
> Why a full get instead of requests.head?

I had two urls fail when using this method, they worked just find with `requests.get` (allow_redirects was enabled).